Problem with new board - Invent One

### Basic Infos

- [X] This issue complies with the [issue POLICY doc](https://github.com/esp8266/Arduino/blob/master/POLICY.md).
- [X] I have read the documentation at [readthedocs](https://arduino-esp8266.readthedocs.io/en/latest) and the issue is not addressed there.
- [X] I have tested that the issue is present in current master branch (aka latest git).
- [X] I have searched the issue tracker for a similar issue.
- [X] If there is a stack dump, I have decoded it.
- [X] I have filled out all fields below.

#### Platform

- Hardware:					ESP-12
- Core Version:			latest git hash or date
- Development Env:		Arduino IDE
- Operating System:		Windows|Ubuntu|MacOS

### Settings in IDE

- Module:  			other
- Flash Mode:			dio
- Flash Size: 		4MB/1MB
- lwip Variant:   v1.4|v2 Lower Memory
- Reset Method:		nodemcu
- Flash Frequency:	40Mhz
- CPU Frequency:		80Mhz
- Upload Using:		SERIAL
- Upload Speed:   115200 (serial upload only)


### Problem Description

Code fails to compile due to clash between inventone analog pin A0 (pins_arduino.h) and the reserved analog pin for arduino (commons.h)

```
#include <Arduino.h>

void setup() {
  Serial.begin(115200);
}

void loop() {
  Serial.println ("Welcome inventone to arduino");
  delay(1000);
}

```

### Debug Messages

```
In file included from /home/jarvis/.arduino15/packages/esp8266/hardware/esp8266/2.5.0/variants/inventone/pins_arduino.h:55:0,
                 from /home/jarvis/.arduino15/packages/esp8266/hardware/esp8266/2.5.0/cores/esp8266/Arduino.h:302,
                 from sketch/sketch_feb08a.ino.cpp:1:
/home/jarvis/.arduino15/packages/esp8266/hardware/esp8266/2.5.0/variants/inventone/../generic/common.h:53:22: error: redefinition of 'const uint8_t A0'
 static const uint8_t A0 = PIN_A0;
                      ^
In file included from /home/jarvis/.arduino15/packages/esp8266/hardware/esp8266/2.5.0/cores/esp8266/Arduino.h:302:0,
                 from sketch/sketch_feb08a.ino.cpp:1:
/home/jarvis/.arduino15/packages/esp8266/hardware/esp8266/2.5.0/variants/inventone/pins_arduino.h:50:22: error: 'const uint8_t A0' previously defined here
 static const uint8_t A0 = 0;
                      ^
exit status 1
Error compiling for board Invent One.
```

--- a/variants/inventone/pins_arduino.h
+++ b/variants/inventone/pins_arduino.h
@@ -45,7 +45,8 @@ static const uint8_t D8   = 12;
 static const uint8_t D9   = 16;
 static const uint8_t RX  = 3;
 static const uint8_t TX = 1;
-static const uint8_t A0 = 0;
+
+#define PIN_A0 0
 /*Analog pins for Onboard ADC*/
 static const uint8_t A1 = 1;
 static const uint8_t A2 = 2;

